### PR TITLE
Check $cur_profile['email_address'] before used

### PR DESCRIPTION
**Pull request by albertlast**
Well to create this issue you had to activate "send_validation_onChange"
but even than i was not fully able to reproduce this error but this should solve this any way.

related issue: https://www.simplemachines.org/community/index.php?topic=559067.0

### CODE REVIEW

File: Sources/Profile-Modify.php [hunk 1]
```diff
@@ -167,7 +167,7 @@ function $preload:		A function that is used to load data required for this eleme
 			'js_submit' => !empty($modSettings['send_validation_onChange']) ? '
 	form_handle.addEventListener(\'submit\', function(event)
 	{
-		if (this.email_address.value != "'. $cur_profile['email_address'] . '")
+		if (this.email_address.value != "'. (!empty($cur_profile['email_address']) ? !empty($cur_profile['email_address']) : '') . '")
```
Comment (Sesquipedalian) [inline Sources/Profile-Modify.php]: I think you probably meant to write this:
```
(!empty($cur_profile['email_address']) ? $cur_profile['email_address'] : '')
```

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
Copy/paste error?

**Review by Sesquipedalian — APPROVED** [review]

**Comment by jdarwood007** [comment]
Whats the error here?  email is a requirement of all profiles, so not sure how you are getting a blank email to trigger this.

**Comment by smfbigguy** [comment]
People were trying to register on my site. I could see them. When they did I would get the error I posted on SMF. After I applied the fix I was able to register perfectly. I have seen this happening on my site for about 4 or 5 days. Numerous people generated that error. That is all I can tell you really.

**Comment by albertlast** [comment]
In this moment when the user accept the licence(in the regestry process) the profile stuff is builded up,
but at this moment the user didn't provide any mail information.

**Pull request merged**